src/Microsoft.SqlTools.ServiceLayer/DacFx/GenerateTSqlModelOperation.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent creating a dacfx model
+    /// </summary>
+    class GenerateTSqlModelOperation
+    {
+        public GenerateTSqlModelParams Parameters { get; }
+
+        public GenerateTSqlModelOperation(GenerateTSqlModelParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Generate model from sql files, if no sql files are passed in then it generates an empty model.
+        /// </summary>
+        public TSqlModel GenerateTSqlModel()
+        {
+            try
+            {
+                TSqlModelOptions options = new TSqlModelOptions();
+                SqlServerVersion version = (SqlServerVersion)Enum.Parse(typeof(SqlServerVersion), Parameters.ModelTargetVersion);
```

Comment:
you should use TryParse instead and return error if fails

Reply:
What are you suggesting here? Because it'll throw an exception regardless. I guess there's some slight benefit to throwing a custom exception here, but given that this is quite the edge case I'm not sure it's worth the extra logic.

Reply:
for extra logging. If ModelTargetVersion has invalid value

Reply:
If the version doesn't exist it'll throw an exception with a message similar to this.

`[System.ArgumentException: Requested value 'Invalid' was not found.]`

I don't think we need to be more explicit, it'll be clear enough from the message what the issue was.

Reply:
(if this was something that we expected to happen often then I'd definitely be in support of being more defensive here. But an invalid version being passed in should not be something we expect to happen normally so I'm not sure the extra logic here would be worth it)